### PR TITLE
softflowd - do not restart on boot plus style fixes

### PR DESCRIPTION
**Pull request by doktornotor**
- Do not try to restart service on boot, otherwise it may get started twice via /etc/rc.start_packages (Fixes bug #4731).
- Add standard XML and copyright headers.
- Remove doubled spaces between sentences in descriptions.
- Developer style guidelines (spacing, braces).

### CODE REVIEW

File: config/softflowd/softflowd.xml [hunk 1]
```diff
@@ -191,39 +243,52 @@
 						"stop" => "/usr/bin/killall -9 softflowd"
 					)
 				);
-				restart_service("softflowd");
+				/* Only try to restart if not booting */
+				if (!($g['booting'])) {
```
Comment (rbgarga) [inline config/softflowd/softflowd.xml]: Please use platform_booting()

File: config/softflowd/softflowd.xml [hunk 2]
```diff
@@ -145,43 +184,56 @@
 		function sync_package_softflowd() {
 			conf_mount_rw();
 			config_lock();
-			global $config;
+			global $config, $g;
```
Comment (rbgarga) [inline config/softflowd/softflowd.xml]: This can be undone now you are using platform_booting()

**Comment by jim-p** [comment]
FYI- platform_booting() does not exist on 2.1.x so this PR broke softflowd for anyone trying to use it on 2.1.x. They should upgrade, undoubtedly, but we haven't officially dropped support for packages on 2.1.x yet so we need to try to not break packages there.

**Comment by doktornotor** [comment]
You guys need to sort this mess yourself. Not to mention - some 70+ package changes have been done here without changing those version 8.x versions at all. Probably time to drop the ball.

**Comment by jim-p** [comment]
Once 2.3 is released only then can we officially drop support for 2.1.x packages -- we typically maintain support for current major version + 1 version back. Ideally fixes should be tested on both and the versions bumped on all affected platforms. The PRs really shouldn't have been approved without taking that into consideration, but in most cases it's fine and the impact is minimal.

If a package is known to be broken on 2.1.x and there is no hope of fixing it, then it can be removed from pkg_config.8.* or have its minimum version raised to 2.2.x.

**Comment by doktornotor** [comment]
TBH: this should have been branched for 2.2. Sounds about two years late for that. The damage done by switching to the PBI shit is beyond repair ATM. Plus, some 1/3 of the packages code is dealing with the crap resulting from that and nothing else.

**Comment by jim-p** [comment]
Yep. Can't wait for 2.3 when a lot of this is left in the dust. Hands are sort of tied until then though. But it's coming fast.

**Comment by doktornotor** [comment]
@jim-p This change has been done in quite a few of packages. Frankly, some of them checking for crap like the CSRF token in order to find whether system is booting or not, I cannot see how's it got more broken that that. If someone's going to sync up the package versions in 8.x and 10.x and let people file bugs about those, it can be dealt with. ATM, 2.1,x is de facto abandoned, no matter the official state.

Finally: Branch all these for 2.3. Point 2.3 package manager to that branch. Then people can clean up all the useless code bloat...

**Comment by rbgarga** [comment]
@doktornotor pfsense-packages repo is not going to be used by 2.3, all packages will be part of pfsense/FreeBSD-ports repository.

**Comment by doktornotor** [comment]
@rbgarga That's a good news.Perhaps, it'd be a good idea to move this debate to the forums. Also, most of these 2.1.x/2.2.x conditional hacks could be removed or simplified a lot if a *final* decision is done. For sure would make the code here less buggy and avoid wasting time with nonsense that's not going to be used anyway.

**Comment by marcelloc** [comment]
What about packages that need non default compile options !?

2015-09-10 14:02 GMT-03:00 doktornotor <notifications@github.com>:

**Comment by rbgarga** [comment]
@marcelloc Options can be found at:

https://github.com/pfsense/pfsense/blob/master/tools/conf/pfPorts/make.conf

**Comment by rbgarga** [comment]
@doktornotor for now, to keep things easy to maintain, there is a script that converts a package from pfsense-packages repo into a port. When 2.3 alpha is ready for public tests we will start these discussions.

**Comment by marcelloc** [comment]
great! Thanks garga.

2015-09-10 14:32 GMT-03:00 Renato Botelho <notifications@github.com>:

**Comment by doktornotor** [comment]
@jim-p - All boot checks done "compatible" against in PR #1059 and PR #1060, kindly review and merge.

**Pull request merged**